### PR TITLE
Use introduced tablet manager RPCs in VTOrc

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR uses the RPCs introduced in https://github.com/vitessio/vitess/pull/10464 in VTOrc.
The RPC `ResetReplicationParameters` is used in `fixPrimaryHasPrimary` to reset the replication parameters for the primary tablet.

The RPC `FullStatus` is used in `ReadTopologyInstanceBufferable` to read information from MySQL server without connecting directly to it.

As part of this refactor, 3 fields from Instance struct have been removed - 
- Uptime :- Unused in the codebase. Used to store the uptime of each mysql instance
- SemiSyncAvailable :- Set when both the semi_sync plugins were loaded. Unused in the codebase and is not required.
- ReplicationCredentialsAvailable :- Unused in the codebase. Used to store if the instance had access to the replication credentials of its replicas

## Related Issue(s)

- #6612 

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by GuptaManan100** [comment]
This PR is based on top of #10464 and should be merged after it

**Review by deepthi — COMMENTED** [review]
This looks fine. I only looked at the orchestrator directory since this PR seems to be on top of #10464.
I can re-review once that is complete and we have a smaller diff in this PR.

File: go/vt/orchestrator/inst/instance_dao.go [hunk 1]
```diff
@@ -299,130 +302,62 @@ func ReadTopologyInstanceBufferable(instanceKey *InstanceKey, bufferWrites bool,
...
-					// I don't want orchestrator to put even more burden on this. The 'Uptime' variable is not that important
-					// so as to completely fail reading a 5.7 instance.
-					// This is supposed to be fixed in 5.7.9
-				}
+		if (instance.IsOracleMySQL() || instance.IsPercona()) && !instance.IsSmallerMajorVersionByString("5.6") {
+			// Stuff only supported on Oracle MySQL >= 5.6
+			// ...
+			// @@gtid_mode only available in Orcale MySQL >= 5.6
+			instance.GTIDMode = fullStatus.GtidMode
+			instance.ServerUUID = fullStatus.ServerUuid
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/orchestrator/inst/instance_dao.go:343]: Ideally, it should be `UUID`. the `Uuid` is coming from the proto file which is generated. Unfortunately, there is no way to change that.

[the remark this answers is not in the document]
Comment (ajm188) [reply go/vt/orchestrator/inst/instance_dao.go:343]: +1 to this (preferring `UUID` except in the case of protobuf-generated code -- vtadmin does similar things)

if you _want_ (read: you should not do this), you can name the field `u_u_i_d` and protobuf will treat those as individual words and you'll get `UUID` in the generated code 🙃 🙃 🙃

[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/orchestrator/inst/instance_dao.go:343]: I was considering `u_u_i_d` too, but then I elected against it 🤣

File: go/test/endtoend/vtorc/readtopologyinstancebufferable/main_test.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
+package readtopologyinstancebufferable
```
Comment (deepthi) [inline go/test/endtoend/vtorc/readtopologyinstancebufferable/main_test.go:1]: nit: can you make this package name a bit shorter?

**Review by deepthi — APPROVED** [review]

File: go/test/endtoend/vtorc/readtopologyinstancebufferable/main_test.go (hunk 2 above)
Comment (GuptaManan100) [reply go/test/endtoend/vtorc/readtopologyinstancebufferable/main_test.go]: Fixed it. Also added the license and the test config file change for running this test.

**Pull request merged**